src/overlays/actors/ovl_En_Fall2/z_en_fall2.c

```diff
@@ -28,26 +35,135 @@ const ActorInit En_Fall2_InitVars = {
+                                     0, 2, 2, 2, 2, 2, 0, 1, 1, 0, 0, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 0, 1,
+                                     1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+    Vtx* vertices = Lib_SegmentedToVirtual(&D_06005F10);
+    u8 perVertexAlphaTable[6];
+    s32 i;
+
+    if (arg0 > 1.0f) {
+        arg0 = 1.0f;
+    }
+    if (arg0 < 0.0f) {
+        arg0 = 0.0f;
+    }
+    perVertexAlphaTable[0] = 0;
+    perVertexAlphaTable[1] = (s8)(54.0f * arg0);
+    perVertexAlphaTable[2] = (s8)(104.0f * arg0);
+    perVertexAlphaTable[3] = (s8)(155.0f * arg0);
+    perVertexAlphaTable[4] = (s8)(255.0f * arg0);
+    perVertexAlphaTable[5] = (s8)(205.0f * arg0);
+
+    for (i = 0; i < ARRAY_COUNT(sAlphaTableIndex); i++, vertices++) {
+        vertices->v.cn[3] = perVertexAlphaTable[sAlphaTableIndex[i]];
+    }
+}
+
+void func_80C1B8A4(EnFall2* this, PlayState* play) {
+}
+
+void func_80C1B8B4(Actor* thisx) {
+    EnFall2* this = THIS;
+    thisx->draw = EnFall2_Draw;
+    if (this->unk2E4 == 1) {
+        Actor_SetScale(thisx, 0.02f);
+    }
+}
+
+void func_80C1B8F0(Actor* thisx) {
+    EnFall2* this = THIS;
+
+    switch (this->unk2E4) {
+        case 1:
+            if (this->unk2E0 < 1.0f) {
+                this->unk2E0 += 0.033333335f;
+            } else {
+                this->unk2E0 = 1.0f;
+            }
+            break;
+        case 2:
+            if (this->unk2E0 > 0.0f) {
+                this->unk2E0 -= 0.1f;
+                if (this->unk2E0 < 0.0f) {
+                    this->unk2E0 = 0.0f;
+                }
+            }
+            if (thisx->scale.x > 0.0f) {
+                thisx->scale.x -= 0.0019999999f;
+                if (thisx->scale.x < 0.0f) {
+                    thisx->scale.x = 0.0f;
+                }
+                thisx->scale.z = thisx->scale.x;
+            }
+    }
+}
+
+void func_80C1B9D4(EnFall2* this, PlayState* play) {
+    func_80183DE0(&this->skeletonInfo);
+    if (Cutscene_CheckActorAction(play, this->unk2E6)) {
+        Cutscene_ActorTranslateAndYaw(&this->actor, play, Cutscene_GetActorActionIndex(play, this->unk2E6));
+        if (this->unk2E4 != play->csCtx.actorActions[Cutscene_GetActorActionIndex(play, this->unk2E6)]->action) {
+            this->unk2E4 = play->csCtx.actorActions[Cutscene_GetActorActionIndex(play, this->unk2E6)]->action;
+            func_80C1B8B4(&this->actor);
+        }
+        func_80C1B8F0(&this->actor);
+        if (this->unk2E0 > 0.0f) {
+            func_800B9010(&this->actor, NA_SE_EV_MOON_LIGHT_PILLAR - SFX_FLAG);
+        }
+    } else {
+        this->actor.draw = NULL;
+    }
+    func_80C1B724(this->unk2E0);
+}
+
+void EnFall2_Update(Actor* thisx, PlayState* play) {
+    EnFall2* this = THIS;
+    this->actionFunc(this, play);
+}
+
+void EnFall2_Draw(Actor* thisx, PlayState* play) {
+    EnFall2* this = THIS;
+    u8 pad;
+    Mtx* mtx;
+
+    if (!(this->unk2E0 <= 0.0f)) {
```

Comment:
Can this be written as `this->unk2E0 > 0.0f`? I know float stuff can be silly sometimes, so it wouldn't surprise me if you *can't*. Worth a shot, though.